presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -242,12 +298,14 @@ public Page getOutput()
 
     @Override
     public void close()
+            throws Exception
     {
         if (!closed) {
             closed = true;
             if (!committed) {
                 pageSink.abort();
             }
+            statisticAggregation.close();
```

Comment:
The problem is that the `operator` is `AutoCloseable`, and not `Closeable`. So i would have to write the cumbersome `Closer` myself. But anyhow, you are right, we need to to this anyway.

Reply:
I implemented a `AutoCloseableCloser`

Reply:
> The problem is that the operator is AutoCloseable, and not Closeable.

This is easy to solve: `closer.register(operator::close)` instead of `closer.register(operator)`.

Reply:
Closeable#close throws the `IOException`, when Autocloseable#close throws a more generic `Exception`